### PR TITLE
Add Non-negative Matrix Factorization model

### PR DESCRIPTION
**Pull request by lthoang**
### Description

Add Non-negative Matrix Factorization model (NMF)

### Checklist:

- [ ] I have added tests.
- [x] I have updated the documentation accordingly.
- [x] I have updated `README.md` (if you are adding a new model).
- [x] I have updated `examples/README.md` (if you are adding a new example).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -158,6 +158,7 @@ The recommender models supported by Cornac are listed below. Why don't you join
 |      | [Matrix Factorization (MF)](cornac/models/mf), [paper](https://datajobs.com/data-science-repo/Recommender-Systems-[Netflix].pdf) | N/A | [biased_mf.py](examples/biased_mf.py), [given_data.py](examples/given_data.py)
 | 2008 | [Collaborative Filtering for Implicit Feedback (CF)](cornac/models/cf), [paper](http://yifanhu.net/PUB/cf.pdf) | [requirements.txt](cornac/models/cf/requirements.txt) | [cf_exp.py](examples/cf_example.py)
 |      | [Probabilistic Matrix Factorization (PMF)](cornac/models/pmf), [paper](https://papers.nips.cc/paper/3208-probabilistic-matrix-factorization.pdf) | N/A | [pmf_ratio.py](examples/pmf_ratio.py)
+| 2001 | [Non-negative Matrix Factorization (NMF)](cornac/models/nmf), [paper](http://papers.nips.cc/paper/1861-algorithms-for-non-negative-matrix-factorization.pdf) | N/A | [nmf_example.py](examples/nmf_example.py)
```
Comment (tqtg) [inline README.md]: Move it down to the end of the list

File: cornac/models/nmf/recom_nmf.pyx [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+                    V_denominator[i, f] += n_ratings * lambda_v * V[i, f] + eps
+                    V[i, f] *= V_numerator[i, f] / V_denominator[i, f]
+
+            # reset all numerators and denominators
+            U_numerator = np.zeros((num_users, num_factors), dtype=np.float32)
+            V_numerator = np.zeros((num_items, num_factors), dtype=np.float32)
+            U_denominator = np.zeros((num_users, num_factors), dtype=np.float32)
+            V_denominator = np.zeros((num_items, num_factors), dtype=np.float32)
+
+            loss = 0.
```
Comment (tqtg) [inline cornac/models/nmf/recom_nmf.pyx]: Why do we need to calculate the loss separately?
Why don't we merge it with the previous loops?

File: docs/source/models.rst [hunk 3]
```diff
@@ -115,6 +115,11 @@ Probabilitic Matrix Factorization (PMF)
 .. automodule:: cornac.models.pmf.recom_pmf
    :members:
 
+Non-negative Matrix Factorization (NMF)
```
Comment (tqtg) [inline docs/source/models.rst]: We need to organize this according to the order of models in the main README file

File: cornac/models/nmf/recom_nmf.pyx (hunk 2 above)
Comment (lthoang) [reply cornac/models/nmf/recom_nmf.pyx]: We cannot calculate loss within prange loop

File: cornac/models/nmf/recom_nmf.pyx [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+                    U[u, f] *= U_numerator[u, f] / U_denominator[u, f]
+
+            # update item factors
+            for i in prange(num_items, nogil=True, num_threads=num_threads):
+                for f in range(num_factors):
+                    loss += lambda_v * V[i, f] * V[i, f]
+                    V_denominator[i, f] += item_counts[i] * lambda_v * V[i, f] + eps
+                    V[i, f] *= V_numerator[i, f] / V_denominator[i, f]
+
+            # reset all numerators and denominators
```
Comment (tqtg) [inline cornac/models/nmf/recom_nmf.pyx]: Is there a better way to reset the values to zero rather than allocating new mem?
For example https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.fill.html

Is it more intuitive to set the values to zero before the loops rather than putting it here?
